Bugfix and new tests pertaining to measurement plugin naming

aepete opened the pull request:
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-labview/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- Updates the list of reserved names which are disallowed as measurement plugin names
  - This constitutes a fix to #187
- Adds new g-test ensuring that attempted usage of a reserved name reports as invalid by `Valid Measurement Service Name.vi`
  - Blank project added as a test asset
- Adds new g-test ensuring that attempted usage of a conflicting name (in case that a library of the requested name already exists in the project) reports as invalid by `Valid Measurement Service Name.vi`
  - Associated project & libraries added as test assets
  
### Why should this Pull Request be merged?

- The bugfix improves product quality, disallowing the user from introducing a namespace conflict via their chosen name
- The tests reduce per-drop & per-release manual test burden, are faster to execute, and cover test cases more exhaustively than we'd typically expect from hand-testing.

### What testing has been done?

- New G tests ran & passed
- Exercised generator dialog, with having attempted conflicting names and reserved names.  Both are rejected, as desired.

dixonjoel commented on Source/Tests/Tests.Generator/Dialog/TestAssets/BlankProject.lvproj:
```diff
@@ -0,0 +1,16 @@
+﻿<?xml version='1.0' encoding='UTF-8'?>
```
Is another option just to have the test script a blank project in a temp directory? What are the advantages of doing it this way?

aepete replied:
You could script it -- and for this particular test that shouldn't be a problem...  But I can anticipate that we might want to execute other tests against a blank project that has disk presence, due to the generator's requirement that it's running in the context of a project that has been saved.  So it just struck me as simplest to put a blank project on-disk as a test asset for all to share.

dixonjoel replied:
Ok. I'm fine with that if we think it has usefulness in the future.

dixonjoel reviewed (approved):
To make sure I understand, the current expectation is for someone to open each of the test VIs by hand and run them before release, correct?

dixonjoel commented:
@aepete If you aren't addressing all of the issues in #165 with this PR, I'd suggest splitting that issue into two so we can close out this part of it.

aepete commented:
> @aepete If you aren't addressing all of the issues in #165 with this PR, I'd suggest splitting that issue into two so we can close out this part of it.

As suggested, forked issue with reserved names into a separate bug (#187), updated #165 and this PR accordingly.
